src/FixedPointNumbers.jl

```diff
@@ -18,27 +18,50 @@ using Compat
 # f => Number of Bytes reserved for fractional part
 abstract FixedPoint{T <: Integer, f} <: Real
 
+
+# Printing. These are used to generate type-symbols, so we need them early.
+function showtype{X<:FixedPoint}(io::IO, ::Type{X})
+    print(io, typechar(X))
+    f = nbitsfrac(X)
+    m = sizeof(X)*8-f-signbits(X)
+    print(io, m, 'f', f)
+    io
+end
+function show{T,f}(io::IO, x::FixedPoint{T,f})
+    showcompact(io, x)
+    showtype(io, typeof(x))
+end
+const _log2_10 = 3.321928094887362
+showcompact{T,f}(io::IO, x::FixedPoint{T,f}) = show(io, round(convert(Float64,x), ceil(Int,f/_log2_10)))
+
 export
     FixedPoint,
     Fixed,
     UFixed,
+# "special" typealiases
     Fixed16,
     UFixed8,
+    U8,
     UFixed10,
     UFixed12,
     UFixed14,
     UFixed16,
-    # literal constructor constants
+    U16,
+    # Q and U typealiases are exported in separate source files
+# literal constructor constants
     uf8,
     uf10,
     uf12,
     uf14,
     uf16,
```

Comment:
If we really don't like these, perhaps we should get rid of them.

But `0xc2N0f8` or `N0f8(0xc2)` can never work as I think you're intending without causing absolute chaos. See https://github.com/JuliaGraphics/ColorTypes.jl/pull/49#issuecomment-247982247. We need to firmly keep in mind the distinction between value and representation of that value, and generally one should just ignore the representation and think about the number in terms of its value. `0xc2 = 194` is far beyond the reach of a number representable with an `N0f8`, so if the constructor doesn't throw an error you've got a big problem.

I think the key is to (1) reliably throw errors when the user tries something that's not allowed (we didn't do that before #48 because of arithmetic overflow), and (2) provide better error messages that actually coach users through how to think about this/fix problems, along the lines of https://github.com/JuliaMath/FixedPointNumbers.jl/pull/49#issue-176928897.